quasar-core/src/main/java/co/paralleluniverse/common/resource/ClassLoaderUtil.java

```diff
@@ -220,13 +212,22 @@ public static InputStream getResourceAsStream(ClassLoader cl, String resource) t
      * @return {@code loader}'s remotest parent that can still find {@code target}.
      */
     public static ClassLoader getBestClassLoader(ClassLoader loader, String resourceName, URL target) {
+        final ClassLoader platformClassLoader = ClassLoader.getPlatformClassLoader();
         ClassLoader current = loader;
-        while (current != null) {
-            final ClassLoader next = current.getParent();
-            final URL resource = getResource(next, resourceName);
+        for (;;) {
+            ClassLoader next = current.getParent();
+            if (next == null) {
+                if (current == platformClassLoader) {
+                    break;
+                }
+                next = platformClassLoader;
+            }
+
+            final URL resource = next.getResource(resourceName);
             if (!target.equals(resource)) {
                 break;
             }
+
             current = next;
```

Comment:
We need a loop exit for this situation:
- `platformClassLoader.getParent()` returns a valid parent
- And that parent can load the resource

Current behaviour is to go to the parents, parent, let's assume it is `null` and then reset `next` to `platformClassLoader` causing a loop with no exit.

It's safe if `ClassLoader.getPlatformClassLoader().getParent()` is guaranteed to return `null` on every JVM. I don't think we have that guarantee.

Reply:
Java's [platform ClassLoader](https://docs.oracle.com/javase/9/migrate/toc.htm#JSMIG-GUID-A868D0B9-026F-4D46-B979-901834343F9E) was introduced in Java 9:
> The extension class loader has been renamed; it is now the platform class loader. All classes in the Java SE Platform are guaranteed to be visible through the platform class loader. In addition, the classes in modules that are standardized under the Java Community Process but not part of the Java SE Platform are guaranteed to be visible through the platform class loader.

In practice, this implies the following:
```java
@Test
public void testJava9() {
    final ClassLoader platformClassLoader = ClassLoader.getPlatformClassLoader();
    final ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();

    assertFalse(platformClassLoader instanceof URLClassLoader);
    assertFalse(systemClassLoader instanceof URLClassLoader);

    assertNull(platformClassLoader.getParent());
    assertEquals(platformClassLoader, systemClassLoader.getParent());
}
```
> It's safe if ClassLoader.getPlatformClassLoader().getParent() is guaranteed to return null on every JVM. I don't think we have that guarantee.

It's true with Oracle JDK, OpenJDK and Azul. Which other JVMs did you have in mind?